### PR TITLE
DSD-1033: MultiSelect final styles

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-1033](https://jira.nypl.org/browse/DSD-1033)

## This PR does the following:

- Updates the styles for the `MultiSelect` component to better adhere to what is shown in the Figma designs.

## How has this been tested?

- local storybook

## Accessibility concerns or updates

- All color combinations adhere to `WCAG 2.1 AA` standards.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/MultiSelect/MultiSelect.stories.mdx [hunk 1]
```diff
@@ -72,23 +75,7 @@ import * as stories from "./MultiSelect.stories.tsx";
...
-
-Resources:
-
-- [downshift.js documentation](https://www.downshift-js.com/)
-- [Reservoir Checkbox Component](http://localhost:6006/?path=/docs/components-form-elements-checkbox--checkbox#accessibility)
-- [W3C WAI Listbox Example](https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/listbox/listbox.html)
-- [W3C WAI on using ARIA role=dialog](https://www.w3.org/WAI/GL/wiki/Using_ARIA_role%3Ddialog_to_implement_a_modal_dialog_box)
-
-## MultiSelect Listbox Variant
+### MultiSelect Listbox Variant
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelect.stories.mdx]: This is now a level 3 heading and under the `Component Props` heading, but the `MultiSelect Dialog Variant` is still a level 2 heading after the `Accessibility` heading. So, this should be updated as well as the table of contents.

File: src/components/MultiSelect/MultiSelectDialog.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { useEffect, useRef, useState } from "react";
 import {
   Box,
   ListItem,
-  Stack,
+  // Stack,
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelectDialog.tsx]: This should be removed.

File: src/components/MultiSelect/MultiSelectDialog.tsx [hunk 3]
```diff
@@ -36,6 +39,7 @@ function MultiSelectDialog({
   const styles = useMultiStyleConfig("MultiSelect", {
     width,
     isBlockElement,
+    isDialog,
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelectDialog.tsx]: Nit: I think you can hardcode this `isDialog: true,` to not have the `isDialog` boolean variable outside of this component.

File: src/components/MultiSelect/MultiSelectDialog.tsx [hunk 4]
```diff
@@ -184,7 +187,8 @@ function MultiSelectDialog({
               ))}
           </UnorderedList>
           {isOpen && !isMobile && (
-            <Stack direction="row" spacing="s" justify="flex-end">
+            // <Stack direction="row" spacing="s" justify="flex-end">
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelectDialog.tsx]: Remove this.

File: src/components/MultiSelect/MultiSelectMenuButton.tsx [hunk 5]
```diff
@@ -88,15 +88,21 @@ const MultiSelectMenuButton = React.forwardRef<
...
             align="right"
             id={`${multiSelectId}-selected-items-count-icon`}
+            marginLeft="xs"
             name="close"
             size="xsmall"
           />
         </Box>
       )}
-      <Box as="span" pr="s">
+      <Box as="span" title={multiSelectLabel} __css={styles.buttonLabel}>
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelectMenuButton.tsx:97]: Why is the `title` attribute set here? Is it for accessibility purposes? Should it use `aria-label` instead?

File: src/theme/components/multiSelect.ts [hunk 6]
```diff
@@ -50,13 +59,41 @@ const MultiSelect = {
       width: "100%",
       // These are for active states
       boxSizing: "border-box",
-      background: "ui.white",
+      // background: "ui.gray.xx-light-cool",
```
Comment (EdwinGuzman) [inline src/theme/components/multiSelect.ts]: Remove this.

File: src/theme/components/multiSelectMenuButton.ts [hunk 7]
```diff
@@ -1,24 +1,59 @@
 const MultiSelectMenuButton = {
-  parts: ["menuButton", "selectedItemsCountButton"],
-  baseStyle: {
+  parts: [
+    "buttonLabel",
+    "menuButton",
+    "selectedItemsCountButton",
+    "toggleIcon",
+  ],
+  // baseStyle: {
```
Comment (EdwinGuzman) [inline src/theme/components/multiSelectMenuButton.ts]: Remove this.

File: src/theme/foundations/radii.ts [hunk 8]
```diff
@@ -1,4 +1,8 @@
 const radii = {
+  button: {
+    default: "2px",
+    pill: "20px",
```
Comment (EdwinGuzman) [inline src/theme/foundations/radii.ts:4]: Why not just use the `pill` key below? Or update any existing radii that use `pill` to now be `button.pill` (such as in the `src/theme/components/button.ts` file line 72) and remove the `pill` statement on line 7.

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Overall looks good but just minor updates and commented out code can be removed. Reviewed this PR with @isastettler .

File: src/components/MultiSelect/MultiSelect.stories.mdx (hunk 1 above)
Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelect.stories.mdx]: Yeah, this component is weird with how the vareiants are broken out.  To make it easier to digest, I will rearrange a few things and add to the descriptions.

File: src/components/MultiSelect/MultiSelectDialog.tsx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelectDialog.tsx]: Isn't that what is done on `line 23`?

File: src/components/MultiSelect/MultiSelectMenuButton.tsx (hunk 5 above)
Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelectMenuButton.tsx:97]: The `title` attribute was introduced to help users when the `MultiSelect` label is truncated.  The truncation is purely handled by styles, so there should be no need for an aria-label with the same thing.  The full text is in there on the button.  It is just not all displayed in the browser for certain situations.

File: src/theme/foundations/radii.ts (hunk 8 above)
Comment (bigfishdesign13) [reply src/theme/foundations/radii.ts:4]: I had visions of creating a separate ticket to handle the change, but I can do it here.

File: src/components/MultiSelect/MultiSelectDialog.tsx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/MultiSelect/MultiSelectDialog.tsx]: Right. I'm saying you don't need that new variable and can have `isDialog: true` here instead, but this is a nitpick.

File: src/components/MultiSelect/MultiSelect.tsx [hunk 9]
```diff
@@ -58,8 +58,11 @@ export type MultiSelectProps = MultiSelectCommonProps &
   MultiSelectVariantsProps;
 
 /**
- * The MultiSelect component is a form input element that presents a list
- * of Checkbox components from which a user can make one or multiple selections.
+ * The `MultiSelect` component is a form input element that presents a list
+ * of `Checkbox` components from which a user can make one or multiple
+ * selections. Two variants of the MultiSelect component are offered, each with
+ * slightly different functionality and requirements.  Because of these
+ * differences, the two vaeraints are broken out in separate stories below.
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelect.tsx]: Typo, "vaeraints"

Comment (EdwinGuzman) [reply src/components/MultiSelect/MultiSelect.tsx]: Did you also mean to say  that they are broken out in "separate stories"? or maybe separate components?

**Review by EdwinGuzman — APPROVED** [review]
Looks good just one minor typo in a comment. You should also connect with Bill before this is merged to avoid significant conflicts.

File: src/components/MultiSelect/MultiSelect.tsx (hunk 9 above)
Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelect.tsx]: Wow, that's a nice typo.

Yes, separate stories on this page.  Ultimately, they are part of the same component.

File: src/components/MultiSelect/MultiSelectDialog.tsx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelectDialog.tsx]: Got it.  I will make that change.

Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelectDialog.tsx]: Actually, this is not even needed anymore.  I originally had different styles for the `listbox` and `dialog` variants, but I finally settled on using the same styles for each.  And with this, there no longer a need to flag the dialog variant.

**Review by nypl-wluisi — APPROVED** [review]

**Pull request merged**